File: crates/auto-lang/test/a2r/017_spec/spec.expected.rs
trait Flyer {
    fn fly(&self);
}


struct Pigeon {}

impl Pigeon {
    fn fly(&self) {
        println!("Flap Flap");
    }
}

impl Flyer for Pigeon {
    fn fly(&self) {
{
            println!("Flap Flap");
        }    }
}

struct Hawk {}

impl Hawk {
    fn fly(&self) {
        println!("Gawk! Gawk!");
    }
}

impl Flyer for Hawk {
    fn fly(&self) {
{
            println!("Gawk! Gawk!");
        }    }
}

fn main() {

    let arr: &[dyn Flyer] = [b1, b2];
    for b in arr {
        b.fly();
    }
}

File: crates/auto-lang/test/a2ts/017_spec/spec.expected.ts
class Pigeon implements Flyer {

    fly(): void {
    console.log("Flap Flap");
}
}

class Hawk implements Flyer {

    fly(): void {
    console.log("Gawk! Gawk!");
}
}

function main(): void {
    

    const b1 = Pigeon();
    const b2 = Hawk();
    



    const arr: Flyer[] = [b1, b2];
    for (const b of arr) {
        b.fly();
    }
}

main();
